NMS-13456: Collectors edits from MMahacek (#3555)
OpenNMS/opennms: docs/modules/operation/pages/performance-data-collection/collectors/snmp/SnmpCollector.adoc

    </snmp-collection>
</datacollection-config>
----
<1> Directory to persist RRD files on the file system.
Ignored if you are using Newts as time-series storage.
<2> Name of the SNMP data collection referenced in the collection package in `collectd-configuration.xml`.
<3> Configure SNMP MIB-II interface metric collection behavior: +
*all* means collect metrics from all interfaces +
*primary* only from interface provisioned as  primary interface +
*select* only from manually selected interfaces from the Web UI.
<4> RRD archive configuration for this set of performance metrics.
Ignored when using Newts as time series storage.
<5> Name of collection group from a file within the `etc/datacollection/` subfolder to include in the collection cycle.

[[ga-performance-management-collectors-snmp-datacollection-configuration]]
.Configuration overview for SNMP data collection
image::performance-management/collectors/01_snmp-datacollection-configuration.png[]

=== SNMP data collection example

> The following scenario steps through creating a new datacollection definition and assumes the following facts:

Reviewer: ```suggestion
The following scenario steps through creating a new datacollection definition and assumes the following:
```

* Node A belongs to categories CatA and CatB, and has a sysObjectID of `.1.100.1.1`.
* Node B belongs to category CatB, and has a sysObjectID of `.1.100.1.2`.

> If you want to collect different OIDs based on a categories filter, this requires you to define separate Collectd packages.

Reviewer: ```suggestion
If you want to collect different OIDs based on a categories filter, you must define separate Collectd packages.
```

    </collect>
    </systemDef>
    <systemDef name="Collect-I">
    <sysoid>.1.100.1.1.2</sysoid>
    <collect>
      <includeGroup>test-group-9</includeGroup>
    </collect>
  </systemDef>
</datacollection-group>
----

There will be two effective snmp-collections called group1 and group2, as the SNMP service appears twice in `collectd-configuration.xml`.
Each one matches a different set of nodes.

Because all the systemDefs have unique names, group1 will contain UniqueA plus UniqueB, meaning it would have Collect-A through Collect-F.
Similarly, group2 would contain Collect-D through Collect-I.
Regardless of the sysoid and sysoidMasks inside the systemDef, what matters at this level is the systemDef name.

*For node A*

Because this node matches two collectd packages for the SNMP service, the collector uses both collections (group1 and group2) and will check Collect-A through Collect-I.
Even though UniqueB is referenced twice, it will only be included once.

Since the node’s sysObjectID is .1.100.1.1, only the systemDefs Collect-A, Collect-C, Collect-D, Collect-F, and Collect-G will be included, as those are the only ones with a sysoid or sysoidMask that matches the sysObjectID.

> The mibObj groups will be added in the order they appear.

Reviewer: ```suggestion
The mibObj groups are added in the order they appear.
```

> If one systemDef references a group name that already included, it won’t be added again.

Reviewer: ```suggestion
If one systemDef references a group name that is already included, it won’t be added again.
```
